File: SCRIPTS/1.8/class_animal_get_set_heranca_polimorfismo.js
class Animal {
    constructor(idade, cor) {
        this._idade = idade
        this._cor = cor
    }

    get idede() {
        return this.idade
    }

    get cor() {
        return this.cor
    }

    set idade(newIdade) {
        this.idade = newIdade
    }

    set cor(newCor) {
        this.cor = newCor
    }

    falar() {}
}

class Cachorro extends Animal {
    constructor(idade, cor) {
        super(idade, cor)
    }

    falar() {
        console.log('Auu! Auu!')
    }
}

class Gato extends Animal {
    constructor(idade, cor) {
        super(idade, cor)
    }

    falar() {
        console.log('Miauu!')
    }
}

let cachorro = new Cachorro(3, 'Gray')
console.log(cachorro)
cachorro.falar()
console.log('--------------------------')

let gato = new Gato(6, 'red')
console.log(gato)
gato.falar()

File: SCRIPTS/1.8/class_animal_get_set_heranca_polimorfismo.py
from abc import ABC, abstractmethod


class Animal(ABC):
    def __init__(self, idade, cor):
        self._idade = idade
        self._cor = cor

    @property
    def idade(self):
        return self._idade

    @property
    def cor(self):
        return self._cor

    @idade.setter
    def idade(self, newIdade):
        self._idade = newIdade

    @cor.setter
    def cor(self, newCor):
        self._cor = newCor

    def falar(self):
        ...
    def

class Cachorro(Animal):
    def __init__(self, idade, cor):
        super().__init__(idade, cor)

    def falar(self):
        print('Falar: Auu! Auu!')


class Gato(Animal):
    def __init__(self, idade, cor):
        super().__init__(idade, cor)

    def falar(self): print('Falar: Miauuu!')


cachorro = Cachorro(2, 'Rex')
cachorro.falar()
print('idade:', cachorro.idade, '\nNome:', cachorro.cor)
print('-------------------------------')
gato = Gato(2, 'Rex')
gato.falar()
print('idade:', gato.idade, '\nNome:', gato.cor)
